### PR TITLE
add cache decorator with timeout

### PR DESCRIPTION
**Pull request by soltys-m**

### CODE REVIEW

File: pragmatic/decorators.py [hunk 1]
```diff
@@ -109,3 +109,72 @@ def wrapper(*args, **kwargs):
...
+
+    @staticmethod
+    def cache_decorator(*args, **kwargs):
+        def _decorator(func):
+            """
+            Decorator to cache return value.
+            """
+
+            # cache_key = kwargs.get('key', f'{func.__qualname__}.{func.__name__}')
+            # cache_version = kwargs.get('version')
```
Comment (eriktelepovsky) [inline pragmatic/decorators.py:154]: ``cache_version = kwargs.get('version', self.cache_version)``

File: pragmatic/decorators.py [hunk 2]
```diff
@@ -109,3 +109,72 @@ def wrapper(*args, **kwargs):
...
+            cache.set(self.key, data, version=self.version, timeout=self.timeout)
+
+    @staticmethod
+    def cache_decorator(*args, **kwargs):
+        def _decorator(func):
+            """
+            Decorator to cache return value.
+            """
+
+            # cache_key = kwargs.get('key', f'{func.__qualname__}.{func.__name__}')
```
Comment (eriktelepovsky) [inline pragmatic/decorators.py:153]: remove comments

File: pragmatic/decorators.py [hunk 3]
```diff
@@ -109,3 +109,72 @@ def wrapper(*args, **kwargs):
...
+
+            @property
+            @wraps(func)
+            def wrapper(self, *args, **kwargs):
+                key = f'{cache_key}.{func.__name__}' if cache_key else f'{self.cache_key}.{func.__name__}'
+                version = self.cache_version
+                cached = cache.get(key, version=version)
+
+                if cached is not None:
+                    # if settings.DEBUG:
```
Comment (eriktelepovsky) [inline pragmatic/decorators.py:166]: remove commented code

Comment (eriktelepovsky) [reply pragmatic/decorators.py:166]: or use logger instead

File: pragmatic/decorators.py (hunk 1 above)
Comment (soltys-m) [reply pragmatic/decorators.py:154]: self is not available on this place

**Pull request merged**